File: problem/Apple_Redistribution/minimumBoxes.py
class Solution:
    def minimumBoxes(self, apple, capacity):
        s = sum(apple)
        freq = [0] * 51
        for c in capacity:
            freq[c] += 1

        ans = 0
        for c in range(50, 0, -1):
            while freq[c] > 0:
                s -= c
                freq[c] -= 1
                ans += 1
                if s <= 0:
                    return ans
        return ans

File: problem/Apple_Redistribution/minimumBoxes.js
/**
 * @param {number[]} apple
 * @param {number[]} capacity
 * @return {number}
 */
var minimumBoxes = function (apple, capacity) {
  let sum = 0;
  for (let i = 0; i < apple.length; i++) sum += apple[i];

  const freq = Array(51).fill(0);
  for (let i = 0; i < capacity.length; i++) freq[capacity[i]]++;

  let ans = 0;
  for (let c = 50; c >= 1; c--) {
    while (freq[c] > 0) {
      sum -= c;
      freq[c]--;
      ans++;
      if (sum <= 0) return ans;
    }
  }
  return ans;
};